MTA-4854 Documentation for Platform Awareness (#181)
migtools/mta-documentation: docs/topics/proc_adding-source-platform-application.adoc

:_newdoc-version: 2.15.0
:_template-generated: 2024-2-21
:_mod-docs-content-type: PROCEDURE

[id="adding-source-platform-application_{context}"]
= Adding a source platform in an application

[role="_abstract"]

> You can associate applications that you created in {ProductShortName} with their Cloud Foundry (CF) instances by configuring the application's coordinates in the source platform. The specific coordinates vary depending on the source platform you select. For example, the coordinates for CF instances are *name* and *space*. 

Reviewer: ```suggestion
You can associate applications that you created in {ProductShortName} with their Cloud Foundry (CF) instances by configuring the application's coordinates in the source platform. 

The specific coordinates vary depending on the source platform you select. For example, the coordinates for CF instances are *name* and *space*. 
```

.Prerequisites

* You configured at least one source platform instance in the *Administration* mode.

.Procedure

. In the {ProductShortName} user interface, select the *Migration* working mode.
. Click *Application Inventory* in the left menu bar.
. Click the edit icon ({icon-edit}). The *Update application* dialog appears.

> . In the form, enter the following information:

Reviewer: ```suggestion
. Enter the following information in the form:
```

Author: Done, thanks

. Click *Source platform* and enter the following fields:
.. *Source platform*: Select a source platform instance.

> .. *Source platform coordinates*: Enable the source platform coordinates to enter application coordinates in JSON. For example, a CF instance coordinates can be:

Reviewer: Is this UI element a checkbox? I got confused by 'enable'. If it is a checkbox, you may use the work 'select the checkbox to enable..'.

+
[source, json]
----
{
    "names": ["app-a", "app-b"],
    "spaces": ["example-space"]
}
----
.. *name*: Type the name of the application.
.. *space*: Configure the CF space in which the app is deployed in the source platform.
. Click *Save* and close the *Update application* dialog.

.Verification

To verify that the coordinates you configured are saved in {ProductShortName}:

> . Click anywhere on the appication row to open the side drawer.

Reviewer: I don't understand what a 'side drawer' is. May be it is my ignorance :). But feel free to replace it with a UI name that is more common, or leave it as is if you think it is commonly used.

:_newdoc-version: 2.15.0
:_template-generated: 2024-2-21
:_mod-docs-content-type: PROCEDURE

[id="adding-source-platform-application_{context}"]
= Adding a source platform in an application

> 

Reviewer 2: The 'Creating a source platform' section should come before all the other sections. The order of the other sections is fine.

[role="_abstract"]
You can associate applications that you created in {ProductShortName} with their Cloud Foundry (CF) instances by configuring the application's coordinates in the source platform. 

The specific coordinates vary depending on the source platform you select. For example, the coordinates for CF instances are *name* and *space*. 

.Prerequisites

* You configured at least one source platform instance in the *Administration* mode.

.Procedure

. In the {ProductShortName} user interface, select the *Migration* working mode.

> . Click *Application Inventory* in the left menu bar.

Reviewer 3: The `Application Inventory` page is opened by default once selecting the "Migration" mode.
I suggest to remove this line and to add to the next step: "On the Application Inventory page.."

. Click the edit icon (image:icon_edit.png[]). The *Update application* dialog appears.
. Enter the following information in the form:
. Click *Source platform* and enter the following fields:
.. *Source platform*: Select a source platform instance.

> .. *Source platform coordinates*: Enable the source platform coordinates to enter application coordinates in JSON. For example, a CF instance coordinates can be:

Reviewer 2: Suggestion:
One could enter application coordinates in JSON format by enabling 'JSON'.

Alternatively, with 'JSON' disabled, once could just type the name of the application deployed on the source platform and the space in which the application is deployed on the source paltform.

+
[source, json]
----

> {
>     "names": ["app-a", "app-b"],
>     "spaces": ["example-space"]
> }

Reviewer 3: In the app instance the source platform coordinates are the name and the space of the app deployed on CF. It should not be a list of apps/spaces
```suggestion
{
  "name": "example-app",
  "space": "example-space"
}
```

----
.. *name*: Type the name of the application.
.. *space*: Configure the CF space in which the app is deployed in the source platform.
. Click *Save* and close the *Update application* dialog.

.Verification

To verify that the coordinates you configured are saved in {ProductShortName}:

> . Click anywhere on the appication row to open the application details view.

Reviewer 3: ```suggestion
. Click anywhere on the appication row to open the application drawer.
```

Author: I am unable to use the word drawer since I must use a few designated words to describe UI in documentation.
